agent/bench-scripts/test-bin/java

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+""" Emulate / mock the behavior of SPECjbb 2005.
+
+The `pbench-specjbb2005` bench-script runs SPECjbb via the `java` command,
+so we provide this fake `java` executable.
+
+It then reads the generated properties file to figure out how to behave, and
+then sets about created the expected output files in the locations determined
+by the properties.
+"""
+import sys
+
+from pathlib import Path
+
+
+if sys.argv[-2] != "-id":
+    sys.exit(0)
```

Comment:
There are two invocations of `java` by `pbench-specjbb2005`, the first needs to succeed silently.  See the `main` branch.

Reply:
But this script should be able to recognize and validate each of them, so that it can tell whether it has been incorrectly invoked.

So, for instance, if `"spec.jbb.Controller" in sys.argv`, then we know that we're in the first invocation, and we can afford to be lenient and just exit successfully.  On the flip side, if `"spec.jbb.JBBmain" in sys.argv`, then we're in the second invocation, and we need to look for `"-id"` and grab the value which follows it in the list.  And, if neither of those tokens is present, we need to complain bitterly.